modules/ingest-common/src/main/java/org/opensearch/ingest/common/RemoveProcessor.java

```diff
@@ -79,6 +80,25 @@ public IngestDocument execute(IngestDocument document) {
                     throw new IllegalArgumentException("field [" + path + "] doesn't exist");
                 }
             }
+            // cannot remove _index, _version and _version_type.
+            if (path.equals(IngestDocument.Metadata.INDEX.getFieldName())
+                || path.equals(IngestDocument.Metadata.VERSION.getFieldName())
+                || path.equals(IngestDocument.Metadata.VERSION_TYPE.getFieldName())) {
+                throw new IllegalArgumentException("cannot remove metadata field [" + path + "]");
+            }
+            // removing _id is disallowed when there's an external version specified in the request
+            if (path.equals(IngestDocument.Metadata.ID.getFieldName())
+                && !document.getFieldValue(IngestDocument.Metadata.VERSION_TYPE.getFieldName(), String.class)
+                    .equals(VersionType.toString(VersionType.INTERNAL))) {
+                Long version = document.getFieldValue(IngestDocument.Metadata.VERSION.getFieldName(), Long.class);
+                String versionType = document.getFieldValue(IngestDocument.Metadata.VERSION_TYPE.getFieldName(), String.class);
```

Comment:
We could probably extract it before `if` statement so we won't be getting the same value twice